### PR TITLE
Optimisations

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What
- Upgrade dp-api-clients-go to v1.12.0. Uses dp-net/http instead of dp-rchttp, with updated default retry counts
- Prevent requests to */latest making a data.json request to Zebedee
- Do not log a failed Zebedee request as an error (the stack trace is excessive for this known case) 
- Only log collection related path when the collection is specified
- Do not log an error if the access_token is not found 

### How to review
- sanity check
- check tests pass

### Who can review
Anyone

### CODE REVIEW

File: middleware/allRoutes/handler.go [hunk 1]
```diff
@@ -40,14 +39,21 @@ func Handler(routesHandler map[string]http.Handler, zebedeeClient *client.Client
 				return
 			}
 
-			// Construct contentPath with any colletion if present in cookie
+			if ok, err := regexp.MatchString(`\/latest$`, req.URL.String()); ok && err == nil {
```
Comment (gedge) [inline middleware/allRoutes/handler.go:42]:
```suggestion
			if strings.HasSuffix(req.URL.String(), "/latest") {
```
hopefully, more efficient than regexp

File: middleware/allRoutes/handler.go [hunk 2]
```diff
@@ -40,14 +40,21 @@ func Handler(routesHandler map[string]http.Handler, zebedeeClient *client.Client
 				return
 			}
 
-			// Construct contentPath with any colletion if present in cookie
+			if strings.HasSuffix(req.URL.String(), "/latest") {
```
Comment (gedge) [inline middleware/allRoutes/handler.go:43]:
```suggestion
			if strings.HasSuffix(path, "/latest") {
```
would be more efficient and also more correct (ignore query string)

Comment (gedge) [inline middleware/allRoutes/handler.go:43]: I would also move this block above the previous two blocks (no point in doing those regex's for these)

Comment (gedge) [reply middleware/allRoutes/handler.go:43]: I wonder if we could change line 36 (the previous `if` block) similarly - i.e. to use `path`

Comment (CarlHembrough) [reply middleware/allRoutes/handler.go]: I think there are cases where the file extension is in the query string. e.g. http://localhost:8888/file?uri=/peoplepopulationandcommunity/personalandhouseholdfinances/incomeandwealth/articles/earlyindicatorestimatesfromthewealthandassetssurvey/attitudestowardssavingforretirementcreditcommitmentsanddebtburdenjuly2014tojune2016/e04e4191.xls

I will be sure to take this point into account for further changes though 👍

Comment (CarlHembrough) [reply middleware/allRoutes/handler.go]: Good point - Done 👍

**Pull request merged**